crates/ruff/src/rules/pyupgrade/rules/outdated_version_block.rs

```diff
@@ -52,36 +52,31 @@ impl BlockMetadata {
     }
 }
 
-fn metadata<T>(locator: &Locator, located: &T) -> Option<BlockMetadata>
+fn metadata<T>(locator: &Locator, located: &T, body: &[Stmt]) -> Option<BlockMetadata>
 where
     T: Ranged,
 {
     indentation(locator, located)?;
 
-    let line_start = locator.line_start(located.start());
-    // Start the selection at the start-of-line. This ensures consistent indentation
-    // in the token stream, in the event that the entire block is indented.
-    let text = locator.slice(TextRange::new(line_start, located.end()));
-
-    let mut starter: Option<Tok> = None;
+    let mut starter = None;
     let mut elif = None;
     let mut else_ = None;
 
-    for (tok, range) in lexer::lex_starts_at(text, Mode::Module, line_start)
-        .flatten()
-        .filter(|(tok, _)| {
-            !matches!(
-                tok,
-                Tok::Indent
-                    | Tok::Dedent
-                    | Tok::NonLogicalNewline
-                    | Tok::Newline
-                    | Tok::Comment(..)
-            )
-        })
+    let body_end = body.last().map(Ranged::end).unwrap();
+    for (tok, range) in lexer::lex_starts_at(
+        locator.slice(located.range()),
+        Mode::Module,
+        located.start(),
+    )
+    .flatten()
+    .filter(|(tok, _)| matches!(tok, Tok::If | Tok::Elif | Tok::Else))
     {
         if starter.is_none() {
             starter = Some(tok.clone());
+        } else if range.start() < body_end {
+            // Continue until the end of the `if` body, thus ensuring that we don't
+            // accidentally pick up an `else` or `elif` in the nested block.
+            continue;
```

Comment:
What if there's an `else` in the `elif` block body? Would we not accidentally flag that as the final `else`?

Reply:
We need `elif` or `else` or neither, We do not need both. So, I simplified the logic.